#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_sparse_constructors.dpatch by Thomas Weber <thomas.weber.mail@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix for 395293, thanks to John Eaton at
## DP: http://www.cae.wisc.edu/pipermail/bug-octave/2006-October/001015.html

@DPATCH@
diff -urNad octave2.9-2.9.9~/src/ov-bool.h octave2.9-2.9.9/src/ov-bool.h
--- octave2.9-2.9.9~/src/ov-bool.h	2006-08-23 18:35:39.000000000 +0000
+++ octave2.9-2.9.9/src/ov-bool.h	2006-10-27 08:26:43.542980411 +0000
@@ -136,6 +136,16 @@
   ComplexNDArray complex_array_value (bool = false) const
     { return ComplexNDArray (dim_vector (1, 1), Complex (scalar)); }
 
+  SparseMatrix sparse_matrix_value (bool = false) const
+    { return SparseMatrix (Matrix (1, 1, scalar)); }
+
+  // XXX FIXME XXX Need SparseComplexMatrix (Matrix) constructor!!!
+  SparseComplexMatrix sparse_complex_matrix_value (bool = false) const
+    { return SparseComplexMatrix (sparse_matrix_value ()); }
+
+  SparseBoolMatrix sparse_bool_matrix_value (bool = false) const
+    { return SparseBoolMatrix (boolMatrix (1, 1, scalar)); }
+
   charNDArray
   char_array_value (bool = false) const
   {
